sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/CheckFileLicenseBase.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using System;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class CheckFileLicenseBase : ParameterLoadingDiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "S1451";
+        protected const string MessageFormat = "Add or update the header of this file.";
+
+        #region HeaderFormat
```

Comment:
Please, do not use regions

Reply:
I'l remove them, but I'd like to understand why. Here, we have 4 highly correlated fields/properties, and my feeling was that putting them into a region made this grouping easier to read.

I've seen people complain that region are bad because they tend to make an excuse for too large classes, and I might even agree, but then the solution is to refactor the code, not simply to remove the region.